src/cluster_legacy.h

```diff
@@ -289,6 +314,24 @@ static_assert(offsetof(clusterMsg, data) == 2256, "unexpected field offset");
                                               primary is up. */
 #define CLUSTERMSG_FLAG0_EXT_DATA (1 << 2) /* Message contains extension data */
 
+typedef struct {
+    char sig[4];     /* Signature "RCmb" (Cluster message bus). */
+    uint32_t totlen; /* Total length of this message */
+    uint16_t ver;    /* Protocol version, currently set to CLUSTER_PROTO_VER. */
+    uint16_t notused1;
+    uint16_t type; /* Message type */
+    union clusterMsgDataLight data;
```

Comment:
let's not leave this to the compiler and the target platform.

```suggestion
    uint16_t type; /* Message type */
    uint16_t notused2;
    union clusterMsgDataLight data;
```

Reply:
Moreover, I think we need two independent concepts

1. whether this is a "light" (again, "compact") message?
2. what payload does it contain?

so here is my new suggestion:

```suggestion
    uint16_t type; /*  CLUSTERMSG_TYPE_LIGHT/COMPACT = 11 */
    uint16_t data_type; /* PUBSUB vs SHARDED PUBSUB, etc */
    union clusterMsgDataLight data;
```

Reply:
We could start light messages from 0x100 or something, so we can have

    #define IS_LIGHT_MSG(type) ((type) >= 0x100)

Reply:
>We could start light messages from 0x100 or something, 

That is a good idea too. Thinking along this line, how about we partition the value space using the MSB, i.e.,

`#define IS_LIGHT_MSG(type) ((type) & 0x8000)` 

or if we are concerned about more flavors of "light" types in the future, we could use the most significant 4 bits instead.

`#define IS_LIGHT_MSG(type) ((type) & 0xF000)` 

That way, we leave enough space between the two types of cluster messages

Reply:
MSB sgtm, 0x8000

Reply:
This might get more complicated wrt to maintaining the stats I think. I will git it a shot